compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -3388,3 +3388,115 @@ OMR::CodeGenerator::redoTrampolineReservationIfNecessary(TR::Instruction *callIn
       self()->fe()->reserveTrampolineIfNecessary(self()->comp(), calleeSymRef, true);
       }
    }
+
+uint32_t
+OMR::CodeGenerator::getCodeSnippetsSize()
+   {
+   uint32_t codeSnippetsSize = 0;
+
+   TR::list<TR::Snippet*> snippetList = self()->getSnippetList();
+
+   for (auto snippets = snippetList.begin(); snippets != snippetList.end(); ++snippets)
+      codeSnippetsSize += (*snippets)->getLength(0);
+
+   return codeSnippetsSize;
+   }
+
+void
+OMR::CodeGenerator::getMethodStats(uint32_t &codeSize, uint32_t &warmBlocks, uint32_t &coldBlocks, uint32_t &prologue,
+                                   uint32_t &snippets, uint32_t &outOfLine, uint32_t &unaccounted,
+                                   uint32_t &blocksInColdCache, uint32_t &overestimateInColdCache)
+   {
+   codeSize = 0;
+   warmBlocks = 0;
+   coldBlocks = 0;
+   prologue = 0;
+   snippets = 0;
+   outOfLine = 0;
+   unaccounted = 0;
+   blocksInColdCache = 0;
+   overestimateInColdCache = 0;  // TODO: implement
+
+   uint32_t allBlocks = 0;
+   uint32_t sizeBeforeFirstBlock = 0;
+   bool firstBlock = true;
+   bool insideColdCache = false;
+   static int const MAX_FREQ = 6;
+   uint32_t cold_frequence_size[MAX_FREQ] = {0};
+
+   codeSize = (uint32_t)(self()->getCodeEnd() - self()->getCodeStart());
+
+#if 0
+   // enable when splitting warm  and cold blocks is enabled
+   if (self()->getLastWarmInstruction())
+      codeSize = codeSize + self()->getWarmCodeEnd() - self()->getColdCodeStart();
+#endif
+
+   const char *names[MAX_FREQ] = {
+     "UNKNOWN_COLD_BLOCK_COUNT",
+     "VERSIONED_COLD_BLOCK_COUNT",
+     "UNRESOLVED_COLD_BLOCK_COUNT",
+     "CATCH_COLD_BLOCK_COUNT",
+     "INTERP_CALLEE_COLD_BLOCK_COUNT",
+     "REVERSE_ARRAYCOPY_COLD_BLOCK_COUNT"
+   };
```

Comment:
Since these are `#define`s in `OMRCfg.hpp`, maybe this array of strings should be defined as a static in `OMR::CFG`? Otherwise there is the risk of the two going out of sync.

Reply:
Agree. Fixed.